File: code_snippets/java/src/main/java/use_cases/WorkflowSubmitter.java
package use_cases;

import dev.restate.sdk.client.Client;
import use_cases.utils.User;

// <start_here>
public class WorkflowSubmitter {
    public void submit(User user){
        // import dev.restate.sdk.client.Client;
        Client rs = Client.connect("http://localhost:8080");
        // mark
        SignupWorkflowClient.fromClient(rs, user.getId())
                // mark
                .submit(user);

        // do something else, with workflow running in the background

        // attach back to the workflow
        // mark
        boolean result = SignupWorkflowClient.fromClient(rs, user.getId())
                // mark
                .workflowHandle()
                // mark
                .attach();
    }
}
// <end_here>

File: code_snippets/kotlin/src/main/kotlin/use_cases/WorkflowSubmitter.kt
package use_cases

import dev.restate.sdk.client.Client

// <start_here>
class WorkflowSubmitter {
    suspend fun submit(user: User) {
        // import dev.restate.sdk.client.Client;
        val rs = Client.connect("http://localhost:8080")
        // mark
        SignupWorkflowClient.fromClient(rs, user.id)
            // mark
            .submit(user)

        // do something else, with workflow running in the background

        // attach back to the workflow
        // mark
        val result: Boolean = SignupWorkflowClient.fromClient(rs, user.id)
            // mark
            .workflowHandle()
            // mark
            .attach()
    }
} // <end_here>